queue/driver/internal.go

```diff
@@ -70,6 +70,23 @@ func (d *Internal) Get(name string) (amboy.Job, error) {
 	return nil, errors.Errorf("no job named %s exists", name)
 }
 
+// Put saves a new job to the queue, returning if it already exists.
+func (d *Internal) Put(j amboy.Job) error {
+	d.jobs.Lock()
+	defer d.jobs.Unlock()
+	name := j.ID()
+
+	_, a := d.jobs.m[name]
+	_, b := d.jobs.dispatched[name]
```

Comment:
Also check pending? I don't fully understand the difference between `d.jobs.m`, `d.jobs.dispatched`, and `d.jobs.pending`.